pandas/tests/extension/test_arrow.py

```diff
@@ -271,6 +284,518 @@ class TestBaseIndex(base.BaseIndexTests):
+        if pa_version_under2p0 and tz not in (None, "UTC"):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason=(f"Not supported by pyarrow < 2.0 with timestamp type {tz}")
+                )
+            )
+        elif (
+            using_array_manager
+            and pa.types.is_duration(data.dtype.pyarrow_dtype)
+            and box_in_series
+        ):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason="Checking ndim when using arraymanager with duration type"
+                )
+            )
+        super().test_setitem_integer_array(data, idx, box_in_series)
+
+    @pytest.mark.parametrize("as_callable", [True, False])
+    @pytest.mark.parametrize("setter", ["loc", None])
+    def test_setitem_mask_aligned(
+        self, data, as_callable, setter, using_array_manager, request
+    ):
+        tz = getattr(data.dtype.pyarrow_dtype, "tz", None)
+        if pa_version_under2p0 and tz not in (None, "UTC"):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason=(f"Not supported by pyarrow < 2.0 with timestamp type {tz}")
+                )
+            )
+        elif using_array_manager and pa.types.is_duration(data.dtype.pyarrow_dtype):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason="Checking ndim when using arraymanager with duration type"
+                )
+            )
+        super().test_setitem_mask_aligned(data, as_callable, setter)
+
+    @pytest.mark.parametrize("setter", ["loc", None])
+    def test_setitem_mask_broadcast(self, data, setter, using_array_manager, request):
+        tz = getattr(data.dtype.pyarrow_dtype, "tz", None)
+        if pa_version_under2p0 and tz not in (None, "UTC"):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason=(f"Not supported by pyarrow < 2.0 with timestamp type {tz}")
+                )
+            )
+        elif using_array_manager and pa.types.is_duration(data.dtype.pyarrow_dtype):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason="Checking ndim when using arraymanager with duration type"
+                )
+            )
+        super().test_setitem_mask_broadcast(data, setter)
+
+    def test_setitem_tuple_index(self, data, request):
+        tz = getattr(data.dtype.pyarrow_dtype, "tz", None)
+        if pa_version_under2p0 and tz not in (None, "UTC"):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason=(f"Not supported by pyarrow < 2.0 with timestamp type {tz}")
+                )
+            )
+        super().test_setitem_tuple_index(data)
+
+    def test_setitem_slice(self, data, box_in_series, using_array_manager, request):
+        tz = getattr(data.dtype.pyarrow_dtype, "tz", None)
```

Comment:
I can try addressing this in a followup, but it's not perfectly generalizable because some of the tests also fail for a specific `unit` or sometimes fails for `UTC`